test/unit/collection_tests.js

```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+const EventEmitter = require('events');
+const chai = require('chai');
+const expect = chai.expect;
+const Db = require('../../lib/db');
+
+class MockTopology extends EventEmitter {
+  constructor() {
+    super();
+  }
+
+  capabilities() {
+    return {};
+  }
+}
+
+describe('Collection', function() {
+  /**
+   * @ignore
+   */
+  it('should not allow atomic operators for findOneAndReplace', {
+    metadata: { requires: { topology: 'single' } },
+    test: function() {
+      const db = new Db('fakeDb', new MockTopology());
+      const collection = db.collection('test');
+      expect(collection.findOneAndReplace.bind(collection, { a: 1 }, { $set: { a: 14 } })).to.throw(
```

Comment:
can you explain why you are using `bind` here, rather than directly calling `collection.findOneAndReplace`?

Reply:
as in the recent [readPreference PR](https://github.com/mongodb/node-mongodb-native/pull/1995/files#diff-da9829d5153566c6bc8607b95771fe48R557), you have to use `bind` because `expect().to.throw()` checks a function, not the result of a function call.

Reply:
Ah! Sorry I should have read more carefully on that PR. I think it shows what we're testing a little more clearly to actually do this in a function:
```
expect(() => {
  collection.findOneAndReplace({ a: 1 }, { $set: { a: 14 } })
}).to.throw(/..../);
```